src/State/State.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useReducer } from 'react';
+import { Todo } from '../types/Todo';
+import { Filter } from '../types/Filter';
+
+type State = {
+  updatedAt: Date;
+  errorMessage: string | null;
+  todos: Todo[];
+  activeTodos: number;
+  clearAll: boolean;
+  filterBy: Filter;
+  isSubmitting: boolean;
+  isEscapeKeyup: boolean;
+  setAllCompleted: boolean,
+  tempTodo: Todo | null,
+};
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export type Action
+  = { type: 'updatedAt' }
+  | { type: 'setError', payload: string | null }
+  | { type: 'clearAll', payload: boolean }
+  | { type: 'saveTodos', payload: { todos: Todo[], activeTodos: number } }
+  | { type: 'setFilter', payload: Filter }
+  | { type: 'setIsSubmitting', payload: boolean }
+  | { type: 'setEscape', payload: boolean }
+  | { type: 'setTodosStatus', payload: boolean }
+  | { type: 'setTempTodo', payload: Todo | null }
+  | { type: 'addActiveTodo', payload: Todo }
+  | { type: 'deleteTodo', payload: number }
+  | { type: 'updateTodo', payload: Todo };
+
+export const initialState: State = {
+  updatedAt: new Date(),
+  errorMessage: null,
+  todos: [],
+  activeTodos: 0,
```

Comment:
no need to save it here - just count the actual number using `todos` (filter it and get length, etc)